atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaHotspottingTest.java

```diff
@@ -34,46 +34,122 @@
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
+        }});
+        return suffersFromIndexHotspoting;
+    }
+
+    private static Schema getIgnoredIndexHotspottingSchema() {
+        Schema ignoredIndexHotspottingSchema = getIgnoredHotspottingSchema();
+        ignoredIndexHotspottingSchema.addIndexDefinition("ignoreHotSpottingIndex", new IndexDefinition(IndexDefinition.IndexType.CELL_REFERENCING) {{
+            ignoreHotspottingChecks();
+            onTable(IGNORED_HOTSPOTTING_TABLE);
+            rowName();
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
+        }});
+        return ignoredIndexHotspottingSchema;
+    }
+
+    private static Schema getIndexFirstRowComponentHashedSchema() {
+        Schema indexFirstRowComponentHashed = getIgnoredHotspottingSchema();
+        indexFirstRowComponentHashed.addIndexDefinition("firstRowComponentHashedIndex", new IndexDefinition(IndexDefinition.IndexType.CELL_REFERENCING) {{
+            onTable(IGNORED_HOTSPOTTING_TABLE);
+            rowName();
+                hashFirstRowComponent();
+                componentFromRow(HOTSPOTTER_ROW_COMPONENT, ValueType.VAR_STRING);
+            rangeScanAllowed();
+        }});
+        return indexFirstRowComponentHashed;
     }
 
     @Test (expected = IllegalStateException.class)
-    public void testHardFailOnValidateOfHotspottingSchema() {
+    public void testHardFailOnValidateOfTableHotspottingSchema() {
         hotspottingSchema.validate();
     }
 
     @Test (expected = IllegalStateException.class)
-    public void testFailToGenerateHotspottingSchema() throws IOException {
+    public void testHardFailOnValidateOfIndexHotspottingSchema() {
+        indexHotspottingSchema.validate();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void testFailToGenerateTableHotspottingSchema() throws IOException {
         hotspottingSchema.renderTables(new TemporaryFolder().getRoot());
     }
 
+    @Test (expected = IllegalStateException.class)
+    public void testFailToGenerateIndexHotspottingSchema() throws IOException {
+        indexHotspottingSchema.renderTables(new TemporaryFolder().getRoot());
+    }
+
     @Test
-    public void testNoFailureWhenHotspottingIgnored() {
+    public void testNoFailureWhenTableHotspottingIgnored() {
         getIgnoredHotspottingSchema().validate();
     }
 
     @Test
-    public void testSuccessfulGenerationWhenHotspottingIgnored() throws IOException {
+    public void testNoFailureWhenIndexHotspottingIgnored() {
+        getIgnoredIndexHotspottingSchema().validate();
+    }
+
+    @Test
+    public void testNoFailureWhenTableFirstRowComponentHashed() {
+        getTableFirstRowComponentHashedSchema().validate();
+    }
+
+    @Test
+    public void testNoFailureWhenIndexFirstRowComponentHashed() {
+        getIndexFirstRowComponentHashedSchema().validate();
```

Comment:
``indexHotspottingSchema`` is used multiple times, so it looked good to have the variable. 
``getIgnoredIndexHotspottingSchema()`` returns schema where hotspotting is ignored while ``indexHotspottingSchema`` hasnt ignored the check.